src/cli/commands/upgrade.js

```diff
@@ -21,12 +34,95 @@ export function hasWrapper(commander: Object, args: Array<string>): boolean {
 export const requireLockfile = true;
 
 export async function run(config: Config, reporter: Reporter, flags: Object, args: Array<string>): Promise<void> {
-  const useLockfile = args.length || flags.latest;
-  const lockfile = useLockfile ? await Lockfile.fromDirectory(config.lockfileFolder, reporter) : new Lockfile();
-  const {dependencies, devDependencies, optionalDependencies, peerDependencies} =
-    (await config.readRootManifest()) || {};
-  const allDependencies = Object.assign({}, peerDependencies, optionalDependencies, devDependencies, dependencies);
-  let addArgs = [];
+  const lockfile = await Lockfile.fromDirectory(config.lockfileFolder);
+  const deps = await getOutdated(config, reporter, flags, lockfile, args);
+
+  // if specific versions were requested for packages, override what getOutdated reported as the latest to install
+  args.forEach(requestedPattern => {
```

Comment:
Is there a reason, or just personal preference? I always visually mistake `for ... of ...` as the old `for ... in ...` which you seldom actually want. The subtle difference always messes me up.